### PR TITLE
REF: give rank1d/2d same nan filling

### PR DESCRIPTION
**Pull request by mzeitlin11**
A piece broken off the `rank_2d` deduplication

### CODE REVIEW

File: pandas/_libs/algos.pyx [hunk 1]
```diff
@@ -931,6 +931,32 @@ ctypedef fused rank_t:
     int64_t
 
 
+cdef rank_t get_rank_nan_fill_val(bint rank_nans_highest, ndarray[rank_t, ndim=1] _):
```
Comment (WillAyd) [inline pandas/_libs/algos.pyx:934]: Does the second argument here need to be an ndarray or can it be a `type`?

**Comment by simonjayhawkins** [comment]
@mzeitlin11 do the stubs need updating for the extra function and extra parameter?

File: pandas/_libs/algos.pyx (hunk 1 above)
Comment (mzeitlin11) [reply pandas/_libs/algos.pyx]: Thanks for the great suggestion! `type` makes sense as a way to allow the `rank_t` parameterization - messed around with this for a while trying to use `type[rank_t]`, but kept running into failures when actually trying to define this value. For example, I wanted to do something like  `cdef type[rank_t] = values.dtype.type`, but kept running into unhelpful errors (maybe this definition just doesn't make sense, but it compiled).

But just pushed a much cleaner solution than the previous way of handling this type specialization.

**Comment by simonjayhawkins** [comment]
> > @mzeitlin11 do the stubs need updating for the extra function and extra parameter?
> 
> This added function (and the function with a new param) will only be called from `Cython` code, so compilation should handle type checking correctness. For another internal function `kth_smallest_c`, I don't see it typed in `algos.pyi`, so I don't think anything needs to be added.



> > @mzeitlin11 do the stubs need updating for the extra function and extra parameter?
> 
> This added function (and the function with a new param) will only be called from `Cython` code, so compilation should handle type checking correctness. For another internal function `kth_smallest_c`, I don't see it typed in `algos.pyi`, so I don't think anything needs to be added.

I see. they are both cdef so not exposed to Python.

File: pandas/_libs/algos.pyx [hunk 2]
```diff
@@ -931,6 +931,32 @@ ctypedef fused rank_t:
     int64_t
 
 
+cdef rank_t get_rank_nan_fill_val(bint rank_nans_highest, rank_t[:] _=None):
```
Comment (jreback) [inline pandas/_libs/algos.pyx:934]: why is there an _ here?

Comment (jreback) [reply pandas/_libs/algos.pyx:934]: can rank_t be required

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/_libs/algos.pyx (hunk 2 above)
Comment (mzeitlin11) [reply pandas/_libs/algos.pyx:934]: The `_` was to signify that the argument is unused. Not requiring `rank_t` here is purposeful - the thought here was that we need an argument with `rank_t` present for fused type specialization, but with it being optional we never have to worry about passing anything.

Comment (jreback) [reply pandas/_libs/algos.pyx:934]: i c, might be worth comment ing in the doc-string as this is non-obvious (oh you did this nvm)

**Review by jreback — APPROVED** [review]

**Pull request merged**